RHEL6_7/system/initscripts/ifcfg/check

```diff
@@ -7,143 +7,157 @@ import re
-	return warning
+    ifcfg_path = "/etc/sysconfig/network-scripts/"
+    ifcfg_prefix = "ifcfg-"
+
+    def get_variable(path, variable):
+        err_msg = "Error while reading ifcfg scripts"
+        def err(output):
+            log_error(err_msg)
+            log_error(output.read())
+
+        try:
+            output = os.tmpfile();
+            (currentpath, _ ) = os.path.split(os.path.realpath(__file__))
+            return_code = subprocess.Popen([currentpath + "/get_var_by_name.sh", path, variable],
+            bufsize=0,
+            executable=None,
+            stdin=None,
+            stdout=output,
+            stderr=subprocess.STDOUT,
+            preexec_fn=None,
+            close_fds=True,
+            shell=False,
+            cwd=None,
+            env=None,
+            universal_newlines=False,
+            startupinfo=None,
+            creationflags=0).wait()
+        except:
+            output.seek(0)
+            err(output)
+            raise
+        output.seek(0)
+        if return_code < 0:#error
+            log_error(output.read())
+            raise err_msg
+        if return_code == 1:#undefined variable
+            return False
+        return output.read().strip()
+
+    def ls_scripts():
+        for f in [i for i in os.listdir(ifcfg_path) if i.startswith(ifcfg_prefix)]:
+            yield f
+
+    def is_kernel(name):
+        return re.match("eth[0-9]+", name)
+
+    def is_udev(name):
+        if name[0:2] in ("en", "sl", "wl", "ww"):
+            if name[2] in ("b", "c", "o", "s", "x", "P", "p"):
+                return True
+        return False
+
+    def fix_ifnames():
+        script_result = subprocess.Popen("./preupg_network.sh", shell=True)
+        script_result.communicate()
+        exit_code = script_result.returncode
+        if exit_code == 1:
+            log_extreme_risk("The network interface names on the source system are in inconsistent state. Upgrade is not possible")
+        elif exit_code == 2:
+            log_extreme_risk("MACADDR directive found in the ifcfg files. Comment it out and rerun preupgrade-assistant. Without that upgrade is not possible")
+
+    ethx_with_addr_count = 0
+    ethx_without_addr_count = 0
+    ifnames_to_fix = 0
+    warning = False
+    for script in ls_scripts():
+        full_path = ifcfg_path + script
+        addr = get_variable(full_path, "HWADDR")
+        name = get_variable(full_path, "NAME" if devel_time else "DEVICE")
+        log_debug("checking " + script + ", name: " + str(name) + " hwaddr: " + str(addr))
+        short_name = script[len(ifcfg_prefix):]
+        if short_name == "lo": #loopback
+            continue
+        if not name:
+            warning = True
+            if not addr:
+                log_slight_risk(full_path + " does not have DEVICE nor HWADDR set, what kind of device is it?")
+            else:
+                log_slight_risk(full_path + " does not have DEVICE set, its name can change after the upgrade.")
```

Comment:
full_path + " does not have DEVICE set, its name can change after the upgrade."
the same as above+what does "its name can change" actually mean?

Reply:
Basically ( just like in the above case) HWADDR and DEVICE directives are normally used by the kernel to determine the mapping of the interface names.  The order in which the interfaces come up is not possible to determine and the interface name CAN change.